### PR TITLE
doc: update nRF RPC UART documentation

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Improve the text formatting and add "Configuration" and "Reliability" sections. Also, remove the information that nRF RPC IPC is used in BLE RPC library, as the latter can now use both IPC and UART transports.

### CODE REVIEW

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 1]
```diff
@@ -10,29 +10,43 @@ nRF RPC UART transport
 The nRF RPC UART transport allows you to use the :ref:`nrf_rpc` library to execute procedures on a remote processor that is connected with the local processor using the UART interface.
 This transport is used in the :ref:`nrf_rpc_ps_samples`.
 
+Configuration
+*************
+
+Set :kconfig:option:`CONFIG_NRF_RPC_UART_TRANSPORT` Kconfig option to enable the nRF RPC UART transport.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:16]:
```suggestion
Enable the :kconfig:option:`CONFIG_NRF_RPC_UART_TRANSPORT` Kconfig option to enable the nRF RPC UART transport.
```

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 2]
```diff
@@ -10,29 +10,43 @@ nRF RPC UART transport
...
+
+Define the ``nordic,rpc-uart`` property in a devicetree overlay file to select the UART peripheral used by the API serialization libraries within |NCS|, such as :ref:`ble_rpc` or :ref:`nfc_rpc`.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:18]: There's two ways to break this sentence into two for readability:
```suggestion
Define the ``nordic,rpc-uart`` property in a devicetree overlay file.
The property is used to select the UART peripheral that is used by the API serialization libraries within the |NCS|, such as :ref:`ble_rpc` or :ref:`nfc_rpc`.
```
or
```suggestion
Define the ``nordic,rpc-uart`` property in a devicetree overlay file to select the UART peripheral.
Select the peripheral used by the API serialization libraries within the |NCS|, such as :ref:`ble_rpc` or :ref:`nfc_rpc`.
```
I think the first option is the correct one, but not sure.

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 3]
```diff
@@ -10,29 +10,43 @@ nRF RPC UART transport
...
-An nRF RPC packet that is sent using the nRF RPC UART transport is encoded within a frame whose format resembles the one used by the HDLC protocol.
-
-Each frame shall start and end with the **delimiter** octet (``0b01111110`` = ``0x7e``).
-Each two subsequent frames may be separated by more than one delimiter octet.
+An nRF RPC packet that is sent using the nRF RPC UART transport is encoded within a frame whose format resembles the one used by the HDLC protocol:
 
-Each byte of the nRF RPC packet shall be encoded according to the following rules:
+1. Each frame shall start and end with the **delimiter** octet (``0b01111110`` = ``0x7e``).
+#. Each two subsequent frames may be separated by more than one delimiter octet.
+#. Each byte of the nRF RPC packet shall be encoded according to the following rules:
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:39]:
```suggestion
* Each frame shall start and end with the **delimiter** octet (``0b01111110`` = ``0x7e``).
* Each two subsequent frames may be separated by more than one delimiter octet.
* Each byte of the nRF RPC packet shall be encoded according to the following rules:
```
These aren't something that happens sequentially, so a numbered list isn't correct here.

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 4]
```diff
@@ -10,29 +10,43 @@ nRF RPC UART transport
...
+     * the **escape** octet (``0x7d``),
+     * the input byte XORed with ``0x20``.
 
-* Otherwise, the byte shall be encoded into a frame’s octet without changes.
+   * Otherwise, the byte shall be encoded into a frame’s octet without changes.
 
-The last two bytes of the frame contain the nRF RPC packet checksum, in little-endian byte order.
-The checksum is calculated using the CRC16_CCITT function with the initial value ``0xffff``.
+#. The last two bytes of the frame contain the nRF RPC packet checksum, in little-endian byte order.
+   The checksum is calculated using the CRC16_CCITT function with the initial value ``0xffff``.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:49]:
```suggestion
* The last two bytes of the frame contain the nRF RPC packet checksum, in little-endian byte order.
  The checksum is calculated using the CRC16_CCITT function with the initial value ``0xffff``.
```

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 5]
```diff
@@ -67,6 +81,23 @@ Then the following octets are transmitted over the UART interface:
 
    7e 80 01 ff 00 00 61 7d 5e f6 6d 72 7e
 
+Reliability
+***********
+
+When :kconfig:option:`CONFIG_NRF_RPC_UART_RELIABLE` is selected, the nRF RPC UART transport enables the reliability feature.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:87]:
```suggestion
When the :kconfig:option:`CONFIG_NRF_RPC_UART_RELIABLE` Kconfig option is selected, the nRF RPC UART transport enables the reliability feature.
```

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 6]
```diff
@@ -67,6 +81,23 @@ Then the following octets are transmitted over the UART interface:
...
+
+1. The receiver of a valid frame acknowledges the frame by replying to the sender with the frame's checksum field.
+#. The sender that has not received an acknowledgment for :kconfig:option:`CONFIG_NRF_RPC_UART_ACK_WAITING_TIME` milliseconds, retransmits the frame.
+   If :kconfig:option:`CONFIG_NRF_RPC_UART_TX_ATTEMPTS` attempts do not result in an acknowledgment, the sender gives up and reports the transmission error.
+#. The frame's checksum field is composed of two values:
+
+   * the most significant bit is the sequence bit that is flipped by the sender for each new transmission.
+   * the remaining bits are 15 least significant bits of the nRF RPC packet checksum.
+
+#. If the received frame has the same checksum field as the previous one, it is rejected as a duplicate.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:99]:
```suggestion
* The receiver of a valid frame acknowledges the frame by replying to the sender with the frame's checksum field.
* If a sender has not received an acknowledgment within a certain time, it retransmits the frame.
  The time (in milliseconds) is defined using the :kconfig:option:`CONFIG_NRF_RPC_UART_ACK_WAITING_TIME` Kconfig option.
* If the sender has not received an acknowledgement after a certain number of attempts, it gives up and reports the transmission error.
  The number of attempts is defined using the :kconfig:option:`CONFIG_NRF_RPC_UART_TX_ATTEMPTS` Kconfig option.
* The frame's checksum field is composed of two values:

  * the most significant bit is the sequence bit that is flipped by the sender for each new transmission.
  * the remaining bits are 15 least significant bits of the nRF RPC packet checksum.

* If the received frame has the same checksum field as the previous one, it is rejected as a duplicate.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst (hunk 1 above)
Comment (peknis) [reply doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:16]: Maybe Use instead of the first Enable.

**Review by peknis — APPROVED** [review]
Just a minor nit to wording proposed by Wille.

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst [hunk 7]
```diff
@@ -8,8 +8,7 @@ nRF RPC IPC Service transport
    :depth: 2
 
 The nRF RPC IPC Service transport allows you to use the :ref:`nrf_rpc` library to execute procedures on remote processors.
-
-This feature is used in the :ref:`ble_rpc` library and also in the :ref:`nrf_rpc_entropy_nrf53` sample.
```
Comment (alxelax) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst:12]: Why did you remove mentioning of `ble_rpc`?

Comment (Damian-Nordic) [reply doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst:12]: Because `ble_rpc` can also use UART, depending on the configuration. Should I rephrase to something like "This feature is used in the :ref:`nrf_rpc_entropy_nrf53` sample and can also be enabled for `ble_rpc`"?

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst [hunk 8]
```diff
@@ -10,29 +10,44 @@ nRF RPC UART transport
 The nRF RPC UART transport allows you to use the :ref:`nrf_rpc` library to execute procedures on a remote processor that is connected with the local processor using the UART interface.
 This transport is used in the :ref:`nrf_rpc_ps_samples`.
 
+Configuration
+*************
+
+Use :kconfig:option:`CONFIG_NRF_RPC_UART_TRANSPORT` Kconfig option to enable the nRF RPC UART transport.
```
Comment (wiba-nordic) [inline doc/nrf/libraries/nrf_rpc/nrf_rpc_uart.rst:16]:
```suggestion
Use the :kconfig:option:`CONFIG_NRF_RPC_UART_TRANSPORT` Kconfig option to enable the nRF RPC UART transport.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst (hunk 7 above)
Comment (alxelax) [reply doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst:12]: Why I would like to leave here any library mentioning (probably it is worth to add nfc and open thread too)? This feature is much wider purposed rather than just sample. Reading this, customer should be aware of abilities and purposes.

Comment (Damian-Nordic) [reply doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst:12]: I personally find it strange to list all components that use a given library. A library is by definition something that is not limited to a specific set of use cases so this always creates a risk that some component is missed and confuses me as a reader: why this doc only mentions A and not B if both A and B use the library in the same way.
But if you disagree, I'll revert the change.

Comment (wiba-nordic) [reply doc/nrf/libraries/nrf_rpc/nrf_rpc_ipc.rst]: We could also split the difference, and have:
"This feature is used in the :ref:`nrf_rpc_entropy_nrf53` sample, and several |NCS| libraries, such as :ref:`ble_rpc`."

By framing the mentioned library (or libraries, if you wish) as examples, it makes it clear the list isn't exhaustive.

**Review by alxelax — APPROVED** [review]
